recipes/msfragger/build.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+MSFRAGGER_SHA256SUM=cdd82dd10c4f3c557805133ea135c58f0c6b90b42f1d792141425b18e75bbdd8
+
+# These values are submitted to the MSFragger site when downloading the application zip.
+if [[ -z "$NAME" ]];
+    then export NAME="${USERNAME:-bioconda}";
+fi
+if [[ -z "$EMAIL" ]];
+    then export EMAIL="${NAME}@${HOSTNAME:-bioconda.org}";
+fi
+if [[ -z "$INSTITUTION" ]];
+    then export INSTITUTION="${HOSTNAME:-bioconda.org}";
+fi
+
+# Create directories
+mkdir -p "$PREFIX/tools"
+#mkdir -p $PREFIX/bin
+
+# Add wrapper python script and link to unprefixed name.
+cp "$RECIPE_DIR/msfragger.py" "$PREFIX/bin"
+ln -s "$PREFIX/bin/msfragger.py" "$PREFIX/bin/msfragger"
+chmod 0755 "${PREFIX}/bin/msfragger"
+
+# This script automates accepting the academic license agreements in order to download MSFragger and build the package. A user of this package is then expected to accept the terms themselves when they download a license key from the MSFragger site, which is enforced by the wrapper script and MSFragger jar file.
+"${RECIPE_DIR}/academic_install.py" -n galaxy -e "$EMAIL" -o "$INSTITUTION" -m "$PKG_VERSION" -p "${PREFIX}/tools"
```

Comment:
All these variables (`EMAIL`, ...) will be `bioconda.org`. Note that this is executed at build time (and not when a user installs it.

Reply:
Just to be sure: To my understanding nobody except the bioconda CI will ever build the package. Users will only get the prebuilt package.

If you want to execute something on install time: https://docs.conda.io/projects/conda-build/en/latest/resources/link-scripts.html or activation time https://docs.conda.io/projects/conda-build/en/latest/resources/activate-scripts.html

Reply:
Right. I agree this code is only meaningful at package build-time. We don't need anything for install or activation time.

I'd also agree the inclusion of these user-specific variables if only really useful if users are building the package themselves outside of bioconda CI. I couldn't have said myself whether that's common, but for example I build it locally during development. In this case I guess it's more accurate, for the MSFragger maintainers, for me to use my own name. It's mostly just logging of downloads for the maintainers, since the end user of the package will need to provide their own key.

But yes, the primary intended outcome that bioconda CI builds this package the end user has nothing to do with this step.